app/services/build_journey_order.rb

```diff
@@ -22,6 +28,17 @@ def entry_lookup_hash
 
   def recursive_path(entry_lookup:, next_entry_id:, entries:)
     entry = entry_lookup.fetch(next_entry_id, nil)
+
+    if entries.include?(entry)
+      send_rollbar_error(message: "A repeated Contentful entry was found in the same journey", entry_id: entry.id)
+      raise RepeatEntryDetected.new(entry.id)
+    end
+
+    if entries.count >= ENTRY_JOURNEY_MAX_LENGTH
+      send_rollbar_error(message: "More than 50 steps were found in a journey map", entry_id: entry.id)
```

Comment:
We could replace the `50` with the value of `ENTRY_JOURNEY_MAX_LENGTH` to have one fewer thing to adjust if needed.

```suggestion
      send_rollbar_error(message: "More than #{ENTRY_JOURNEY_MAX_LENGTH} steps were found in a journey map", entry_id: entry.id)
```